src/Type/CommonUnionType.php

```diff
@@ -122,4 +123,19 @@ public function changeBaseClass(string $className): StaticResolvableType
 		return new self(UnionTypeHelper::changeBaseClass($className, $this->getTypes()));
 	}
 
+	public function isIterable(): int
+	{
+		return self::RESULT_NO;
+	}
+
+	public function getIterableKeyType(): Type
+	{
+		return new MixedType();
```

Comment:
The `\PHPStan\Type\CommonUnionType::isIterable()` is a bit problematic as it currently intentionally lies as returning the correct value breaks assumption done by code in few places (as UnionIterableType is expected to be iterable and CommonUnionType is expected to not be iterable even though it contains iterable values inside).

Returning here `ErrorType` is problematic as it will give user false positives in some cases (when iterating over CommonUnionType).

Returning MixedType matches current behavior where iterating over value that is not-iterable results in mixed values, i.e. it is the most compatible way to handle this now. Future PR are expected to improve upon this behavior.